joplin/api/schema.py

```diff
@@ -6,19 +6,116 @@
+from wagtail.core.blocks import *
 from wagtail.documents.models import Document
-
+from wagtail.core.rich_text import expand_db_html
 from base.models import TranslatedImage, ThreeOneOne, ServicePage, ServicePageContact, ServicePageTopic, ServicePageRelatedDepartments, InformationPageRelatedDepartments, ProcessPage, ProcessPageStep, ProcessPageContact, ProcessPageTopic, InformationPage, InformationPageContact, InformationPageTopic, DepartmentPage, DepartmentPageContact, DepartmentPageDirector, Theme, TopicCollectionPage, TopicPage, Contact, Location, ContactDayAndDuration, Department, DepartmentContact, TopicPageTopicCollection, OfficialDocumentPage, OfficialDocumentPageRelatedDepartments, OfficialDocumentPageTopic, OfficialDocumentPageOfficialDocument, GuidePage, GuidePageTopic, GuidePageRelatedDepartments, GuidePageContact, JanisBasePage, PhoneNumber, DepartmentPageTopPage, DepartmentPageRelatedPage
+import collections
+import traceback
+from nested_lookup import nested_lookup, nested_alter, nested_update, get_occurrence_of_key, get_occurrence_of_value
+
+
+class RichTextFieldType(Scalar):
+    """Serialises RichText content into fully baked HTML
+    see https://github.com/wagtail/wagtail/issues/2695#issuecomment-373002412 """
+
+    @staticmethod
+    def serialize(value):
+        return expand_db_html(value)
+
+
+@convert_django_field.register(RichTextField)
+def convert_rich_text_field(field, registry=None):
+    return RichTextFieldType(
+        description=field.help_text, required=not field.null
+    )
+
+
+def find_rich_text_names(StreamValue):
+    """
+    returns a list of keys of StreamField blocks that are of RichText type
+    these are the ones we need to expand_db_html on
+    TODO: this dosen't actually get us all the values we need to modify 
+    """
+    rich_text_names = []
+    rich_text_names.append('options')
+    for item in StreamValue:
+        block = item.block
+        try:
+            if isinstance(block, RichTextBlock):
+                rich_text_names.append(block.name)
+            elif block.child_blocks:
+                for child_block in block.all_blocks():
+                    if isinstance(child_block, RichTextBlock):
+                        rich_text_names.append(child_block.name)
+        except AttributeError as e:
+            pass
+    return rich_text_names
 
 
 class StreamFieldType(Scalar):
+    """
+    todo
+            for item in serialized:
+                loop through all the values and try running expand_db_html on them to return an href
+                still need to find a way to actually attach the janis_url to the selection tho..
+    """
+
     @staticmethod
-    def serialize(dt):
-        return [{'type': item.block_type, 'value': item.block.get_api_representation(item.value), 'id': item.id} for item in dt]
+    def serialize(StreamValue):
+        """
+        This is a rats nest of loops and conditionals, but the general idea is two functions:
```

Comment:
It's quite hard to parse right now, but could be pretty easily cleaned up by moving a couple chunks into named methods.